`core0.test_dlmalloc_inline` regression

Something all my bots are currently hitting, which I first thought was an out-of-date cache or an out-of-date incremental LLVM build.. but is now persisting after a full clean, is

```
C:\emsdk\emscripten\main>test\runner test_dlmalloc_inline
Running 1 tests
[1/1] test_dlmalloc_inline (test_core.core0.test_dlmalloc_inline) ... FAIL

Stdout:
src.c:887:50: error: alias and aliasee have different types 'void *(size_t)' (aka 'void *(unsigned long)') and 'void *(size_t, size_t)' (aka 'void *(unsigned long, unsigned long)') [-Werror,-Wattribute-alias]
  887 | void* __libc_calloc(size_t) __attribute__((weak, alias("dlcalloc")));
      |                                                  ^
src.c:4885:7: note: aliasee is declared here
 4885 | void* dlcalloc(size_t n_elements, size_t elem_size) {
      |       ^
src.c:906:61: error: alias and aliasee have different types 'size_t (const void *)' (aka 'unsigned long (const void *)') and 'size_t (void *)' (aka 'unsigned long (void *)') [-Werror,-Wattribute-alias]
  906 | size_t malloc_usable_size(const void*) __attribute__((weak, alias("dlmalloc_usable_size")));
      |                                                             ^
src.c:5486:8: note: aliasee is declared here
 5486 | size_t dlmalloc_usable_size(void* mem) {
      |        ^
2 errors generated.
emcc: error: ''C:/emsdk/llvm/git/build_main_vs2022_64/Release/bin\clang.exe' -target wasm32-unknown-emscripten -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr '--sysroot=C:\emsdk\emscripten\main\cache\sysroot' -Xclang '-iwithsysroot/include\fakesdl' -Xclang '-iwithsysroot/include\compat' -Werror -O0 -c src.c -o 'C:\Users\clb\AppData\Local\Temp\emscripten_temp_nej_e824\src.o'' failed (returned 1)

None


======================================================================
FAIL: test_dlmalloc_inline (test_core.core0.test_dlmalloc_inline)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\common.py", line 1211, in run_process
    rtn = utils.run_process(cmd, check=check, **kwargs)
  File "C:\emsdk\emscripten\main\tools\utils.py", line 45, in run_process
    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
  File "C:\emsdk\python\3.13.3_64bit\Lib\subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\emsdk\\emscripten\\main\\emcc.bat', 'src.c', '-o', 'src.js', '-sNO_DEFAULT_TO_CXX', '-sINITIAL_MEMORY=128mb', '-Wclosure', '-Werror', '-Wno-limited-postlink-optimizations', '-O0']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\test_core.py", line 338, in decorated
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 353, in decorated
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_core.py", line 6479, in test_dlmalloc_inline
    self.do_run(src, '*1,0*', args=['200', '1'], force_c=True)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1389, in do_run
    return self._build_and_run(filename, expected_output, **kwargs)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1421, in _build_and_run
    js_file = self.build(filename, **kwargs)
  File "C:\emsdk\emscripten\main\test\common.py", line 911, in build
    self.run_process(cmd, stderr=self.stderr_redirect if not DEBUG else None)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\emsdk\emscripten\main\test\common.py", line 1216, in run_process
    self.fail(f'subprocess exited with non-zero return code({e.returncode}): `{shlex.join(cmd)}`')
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: subprocess exited with non-zero return code(1): `'C:\emsdk\emscripten\main\emcc.bat' src.c -o src.js -sNO_DEFAULT_TO_CXX -sINITIAL_MEMORY=128mb -Wclosure -Werror -Wno-limited-postlink-optimizations -O0`

Stdout:
src.c:887:50: error: alias and aliasee have different types 'void *(size_t)' (aka 'void *(unsigned long)') and 'void *(size_t, size_t)' (aka 'void *(unsigned long, unsigned long)') [-Werror,-Wattribute-alias]
  887 | void* __libc_calloc(size_t) __attribute__((weak, alias("dlcalloc")));
      |                                                  ^
src.c:4885:7: note: aliasee is declared here
 4885 | void* dlcalloc(size_t n_elements, size_t elem_size) {
      |       ^
src.c:906:61: error: alias and aliasee have different types 'size_t (const void *)' (aka 'unsigned long (const void *)') and 'size_t (void *)' (aka 'unsigned long (void *)') [-Werror,-Wattribute-alias]
  906 | size_t malloc_usable_size(const void*) __attribute__((weak, alias("dlmalloc_usable_size")));
      |                                                             ^
src.c:5486:8: note: aliasee is declared here
 5486 | size_t dlmalloc_usable_size(void* mem) {
      |        ^
2 errors generated.
emcc: error: ''C:/emsdk/llvm/git/build_main_vs2022_64/Release/bin\clang.exe' -target wasm32-unknown-emscripten -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr '--sysroot=C:\emsdk\emscripten\main\cache\sysroot' -Xclang '-iwithsysroot/include\fakesdl' -Xclang '-iwithsysroot/include\compat' -Werror -O0 -c src.c -o 'C:\Users\clb\AppData\Local\Temp\emscripten_temp_nej_e824\src.o'' failed (returned 1)

None

----------------------------------------------------------------------
Ran 1 test in 0.441s

FAILED (failures=1)
```

Oddly, I was not getting this failure on my Windows dev PC at first, until I updated LLVM build to latest (with `emsdk install sdk-main-64bit` to compile LLVM from source). So this suggests this would be some kind of an issue on LLVM side?

The errors look valid though - so maybe a new diagnostics item has been added to LLVM? Or I wonder if I've still got something stale.

--- a/system/lib/dlmalloc.c
+++ b/system/lib/dlmalloc.c
@@ -884,7 +884,7 @@ extern "C" {
 #if defined(__EMSCRIPTEN__)
 void* __libc_malloc(size_t) __attribute__((weak, alias("dlmalloc")));
 void  __libc_free(void*) __attribute__((weak, alias("dlfree")));
-void* __libc_calloc(size_t) __attribute__((weak, alias("dlcalloc")));
+void* __libc_calloc(size_t, size_t) __attribute__((weak, alias("dlcalloc")));
 void* __libc_realloc(void*, size_t) __attribute__((weak, alias("dlrealloc")));
 void* malloc(size_t) __attribute__((weak, alias("dlmalloc")));
 void  free(void*) __attribute__((weak, alias("dlfree")));
@@ -903,7 +903,7 @@ int malloc_trim(size_t) __attribute__((weak, alias("dlmalloc_trim")));
 #if !NO_MALLOC_STATS
 void malloc_stats(void) __attribute__((weak, alias("dlmalloc_stats")));
 #endif
-size_t malloc_usable_size(const void*) __attribute__((weak, alias("dlmalloc_usable_size")));
+size_t malloc_usable_size(void*) __attribute__((weak, alias("dlmalloc_usable_size")));
 size_t malloc_footprint(void) __attribute__((weak, alias("dlmalloc_footprint")));
 size_t malloc_max_footprint(void) __attribute__((weak, alias("dlmalloc_max_footprint")));
 size_t malloc_footprint_limit(void) __attribute__((weak, alias("dlmalloc_footprint_limit")));

--- a/test/codesize/test_codesize_cxx_ctors1.json
+++ b/test/codesize/test_codesize_cxx_ctors1.json
@@ -2,9 +2,9 @@
   "a.out.js": 19322,
   "a.out.js.gz": 8008,
   "a.out.nodebug.wasm": 132627,
-  "a.out.nodebug.wasm.gz": 49918,
+  "a.out.nodebug.wasm.gz": 49916,
   "total": 151949,
-  "total_gz": 57926,
+  "total_gz": 57924,
   "sent": [
     "__cxa_throw",
     "_abort_js",

--- a/test/codesize/test_codesize_cxx_ctors2.json
+++ b/test/codesize/test_codesize_cxx_ctors2.json
@@ -2,9 +2,9 @@
   "a.out.js": 19299,
   "a.out.js.gz": 7995,
   "a.out.nodebug.wasm": 132053,
-  "a.out.nodebug.wasm.gz": 49575,
+  "a.out.nodebug.wasm.gz": 49567,
   "total": 151352,
-  "total_gz": 57570,
+  "total_gz": 57562,
   "sent": [
     "__cxa_throw",
     "_abort_js",

--- a/test/codesize/test_codesize_cxx_except.json
+++ b/test/codesize/test_codesize_cxx_except.json
@@ -2,9 +2,9 @@
   "a.out.js": 23298,
   "a.out.js.gz": 8987,
   "a.out.nodebug.wasm": 172519,
-  "a.out.nodebug.wasm.gz": 57443,
+  "a.out.nodebug.wasm.gz": 57432,
   "total": 195817,
-  "total_gz": 66430,
+  "total_gz": 66419,
   "sent": [
     "__cxa_begin_catch",
     "__cxa_end_catch",

--- a/test/codesize/test_codesize_cxx_except_wasm.json
+++ b/test/codesize/test_codesize_cxx_except_wasm.json
@@ -2,9 +2,9 @@
   "a.out.js": 19125,
   "a.out.js.gz": 7925,
   "a.out.nodebug.wasm": 147919,
-  "a.out.nodebug.wasm.gz": 55314,
+  "a.out.nodebug.wasm.gz": 55312,
   "total": 167044,
-  "total_gz": 63239,
+  "total_gz": 63237,
   "sent": [
     "_abort_js",
     "_tzset_js",

--- a/test/codesize/test_codesize_cxx_except_wasm_legacy.json
+++ b/test/codesize/test_codesize_cxx_except_wasm_legacy.json
@@ -2,9 +2,9 @@
   "a.out.js": 19199,
   "a.out.js.gz": 7955,
   "a.out.nodebug.wasm": 145725,
-  "a.out.nodebug.wasm.gz": 54936,
+  "a.out.nodebug.wasm.gz": 54928,
   "total": 164924,
-  "total_gz": 62891,
+  "total_gz": 62883,
   "sent": [
     "_abort_js",
     "_tzset_js",

--- a/test/codesize/test_codesize_cxx_mangle.json
+++ b/test/codesize/test_codesize_cxx_mangle.json
@@ -2,9 +2,9 @@
   "a.out.js": 23348,
   "a.out.js.gz": 9009,
   "a.out.nodebug.wasm": 238948,
-  "a.out.nodebug.wasm.gz": 79814,
+  "a.out.nodebug.wasm.gz": 79805,
   "total": 262296,
-  "total_gz": 88823,
+  "total_gz": 88814,
   "sent": [
     "__cxa_begin_catch",
     "__cxa_end_catch",

--- a/test/codesize/test_codesize_cxx_noexcept.json
+++ b/test/codesize/test_codesize_cxx_noexcept.json
@@ -2,9 +2,9 @@
   "a.out.js": 19322,
   "a.out.js.gz": 8008,
   "a.out.nodebug.wasm": 134643,
-  "a.out.nodebug.wasm.gz": 50760,
+  "a.out.nodebug.wasm.gz": 50756,
   "total": 153965,
-  "total_gz": 58768,
+  "total_gz": 58764,
   "sent": [
     "__cxa_throw",
     "_abort_js",

